Applications/FileIO/SWMM/SWMMInterface.h

```diff
@@ -0,0 +1,245 @@
+
+    /// If a mesh has already been created, this methods allows to add node- or link-arrays as property
+    /// to that mesh. The data vectors can be created using the getArrayAtTimeStep() method.
+    static bool addResultsToMesh(MeshLib::Mesh &mesh, SwmmObject const type,
+        std::string const& vec_name, std::vector<double> const& data);
+
+    /// Returns the mesh generated from SWMM file content.
+    MeshLib::Mesh* getMesh() const { return _mesh.get(); }
+
+    /// Returns the name of the data array for the given object type and parameter index.
+    std::string getArrayName(SwmmObject obj_type, std::size_t var_idx) const;
+
+    /// Get all the object names for a given object type
+    std::vector<std::string> getNames(SwmmObject obj_type) const;
+
+    /// Returns the Name for the indexed object of the given type (or an empty string if an error occured).
+    std::string getName(SwmmObject obj_type, std::size_t idx) const;
+
+    /// Returns the number of objects of the given type.
+    std::size_t getNObjects(SwmmObject obj_type) const;
+
+    /// Returns the number of parameters (incl. pollutants) of the given type.
+    std::size_t getNParameters(SwmmObject obj_type) const;
+
+    /// Returns the number of time steps for the simulation results.
+    std::size_t getNTimeSteps() const;
+
+    /// Returns an array for a given variable at all nodes/links from a SWMM output file for a given time step.
+    std::vector<double> getArrayAtTimeStep(SwmmObject obj_type, std::size_t time_step, std::size_t var_idx) const;
+
+    /// Returns an array for a given variable for one specific object from a SWMM output file for all time steps.
+    std::vector<double> getArrayForObject(SwmmObject obj_type, std::size_t obj_idx, std::size_t var_idx) const;
+
+    /// Writes the node coordinates into double vectors for writing of CSV files
+    bool getNodeCoordinateVectors(std::vector<double> &x, std::vector<double> &y, std::vector<double> &z) const;
+
+    /// Writes the inlet- and outlet IDs for links into vectors for writing of CSV files
+    bool getLinkPointIds(std::vector<std::size_t> &inlets, std::vector<std::size_t> &outlets) const;
+
+    /// Write a CSV file for all object of the given type at one time step
+    bool writeCsvForTimestep(std::string const& file_name, SwmmObject obj_type, std::size_t time_step) const;
+
+    /// Write a CSV file for one object of the given type for all time steps
+    bool writeCsvForObject(std::string const& file_name, SwmmObject obj_type, std::size_t obj_idx) const;
+
+    /// Checks if file is a SWMM input file
+    static bool isSwmmInputFile(std::string const& inp_file_name);
+
+    /// Checks if file is a SWMM output file
+    static bool isSwmmOutputFile(std::string const& inp_file_name);
+
+    /// Reading a SWMM input file and conversion into OGS geometry.
+    static bool SwmmInputToGeometry(std::string const& inp_file_name, GeoLib::GEOObjects &geo_objects, bool add_subcatchments);
```

Comment:
Long line. Please break.
The method name is "unusual": No verb and starting uppercase.